addons/project/models/project.py

```diff
@@ -1198,13 +1329,23 @@ def create(self, vals_list):
                 rec_values['next_recurrence_date'] = fields.Datetime.today()
                 recurrence = self.env['project.task.recurrence'].create(rec_values)
                 vals['recurrence_id'] = recurrence.id
-        tasks = super().create(vals_list)
+        if portal_can_create:
+            self = self.sudo()
+        tasks = super(Task, self).create(vals_list)
         for task in tasks:
             if task.project_id.privacy_visibility == 'portal':
                 task._portal_ensure_token()
         return tasks
 
     def write(self, vals):
+        portal_can_create = False
+        if self.env.user.has_group('base.group_portal') and not self.env.su:
+            # Check if all fields in vals are in SELF_WRITABLE_FIELDS
+            self._ensure_portal_user_can_write(vals.keys())
+            self.check_access_rights('write')
+            self.check_access_rule('write')
+            portal_can_create = True
```

Comment:
hmm, what tells you that task is part of a project shared with the user?

Reply:
and that seems to mean `portal_can_write` rather than `portal_can_create`, no?

Reply:
The call to ``check_access_rule('write')``